tests-wtr/dom-actions.test.ts

```diff
@@ -0,0 +1,137 @@
+import { expect } from '@esm-bundle/chai';
+import {DomActions} from "../lib/dom-actions";
+
+function instantiateDomActions() {
+  return new DomActions({config: {logs: {rulesteps: false, lifecycle: false, evals: false, errors: false, messages: false}}})
+}
+
+function getEventCounterFromDataset(eventType: string, querySelector: string, doc: Document | ShadowRoot = document) {
+  return parseInt(doc.querySelector<HTMLElement>(querySelector).dataset[`event:${eventType}`] ?? '0', 10)
+}
+
+function getClicksCount(querySelector: string, doc: Document | ShadowRoot = document) {
+  return getEventCounterFromDataset('click', querySelector, doc)
+}
+
+function addListenerThatIncrementsInDataset(el: HTMLElement, eventType: string) {
+  const listener = () => {
+    const current = parseInt(el.dataset[`event:${eventType}`] ?? '0', 10)
+
+    el.dataset[`event:${eventType}`] = (current + 1).toString()
```

Comment:
I went with a bit different approach than the original one - instead of storing the values in `.innerText`, I used `dataset` as it felt a bit more "meta" to me. But could be the `.innerText` approach too - the advantage of that might be that when debugging with `--manual` mode, it would be visible at first sight, unlike `dataset`. I didn't have any strong reasons for this, so we can revert to using innerText as previously. :)

Reply:
oh wow brilliant idea!! Much simpler indeed, love it ❤️